rosbag2_transport/include/rosbag2_transport/recorder.hpp

```diff
@@ -109,12 +109,13 @@ class Recorder : public rclcpp::Node
 
   inline constexpr static const auto kPauseResumeToggleKey = KeyboardHandler::KeyCode::SPACE;
 
-private:
-  void topics_discovery();
-
+protected:
   std::unordered_map<std::string, std::string>
   get_requested_or_available_topics();
```

Comment:
@theseankelly Windows build fails with linker error:
```
12:49:07 test_record_all_include_unpublished_topics.obj : error LNK2019: unresolved external symbol "protected: class std::unordered_map<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> >,struct std::hash<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >,struct std::equal_to<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > >,class std::allocator<struct std::pair<class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const ,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > > > > __cdecl rosbag2_transport::Recorder::get_requested_or_available_topics(void)"
```
It seems to fix this linker error need to add 
ROSBAG2_TRANSPORT_EXPORT before declaration of the `get_requested_or_available_topics(void)`